code/script.js

```diff
@@ -244,100 +296,155 @@ const generateBoard = () => {
+// (re)set the board,
+// (re)set the guess list,
+// invoke the generation of characters,
+// and invoke the selection of secret character
 const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  board.style.display = 'flex'
+  winOrLose.style.display = 'none'
+  questions.selectedIndex = '0'
+  generateBoard()
+  setSecret()
+  // console.log(secret)
 }
 
-// setting the currentQuestion object when you select something in the dropdown
+// Sets & stores the values from the drop-down,
+// to be used when user guesses
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-
-  if (category === 'hair color') {
+  const guessCategory = questions.options[questions.selectedIndex].parentNode.label
+  const guessValue = questions.value
+  // console.log(guessValue)
+  // console.log(guessCategory)
+  if (guessCategory === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
-      category: category,
+      value: guessValue,
+      category: guessCategory,
     }
-  } else if (category === 'eye color') {
-    // Set this up your self
-  } else if (category === 'accessories') {
-    currentQuestion = {
-      //attribute: ,
-      // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
-      value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
-      category: category,
+  } else if (guessCategory === 'eye color') {
+      currentQuestion = {
+        attribute: 'eyeColor',
+        value: guessValue,
+        category: guessCategory,
+      }
+  } else if (guessCategory === 'accessories') {
+      currentQuestion = {
+        attribute: guessValue,
+        value: true,
+        category: guessCategory,
     }
-  } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
+  } else if (guessCategory === 'other') {
+      currentQuestion = {
+        attribute: guessValue,
+        value: true,
+        category: guessCategory,
+      }
   }
 }
 
-// This function should be invoked when you click on 'Find Out'.
+// When the user takes a guess guess,
+// compare the guess to the answer,
+// and pass information along to the filter function
 const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+  const guess = currentQuestion.value === secret[currentQuestion.attribute]
+  // console.log(guess)
+  filterCharacters(guess)
```

Comment:
I like that you are renaming the variables! Since this is a boolean, it could be named something like `isGuessCorrect` or something to indicate that its value is either true or false